src/classes/assets.py

```diff
@@ -37,15 +37,24 @@ def get_assets_path(file_path=None, create_paths=True):
         return info.USER_PATH
 
     try:
-        # Generate asset folder name, max 30 chars of filename + "_assets"
+        # Generate asset folder name filename + "_assets"
         file_path = file_path
         asset_filename = os.path.splitext(os.path.basename(file_path))[0]
-        asset_folder_name = asset_filename[:30] + "_assets"
+        asset_folder_name = asset_filename + "_assets"
         asset_path = os.path.join(os.path.dirname(file_path), asset_folder_name)
 
+        # Previous Assets File Name Convention.
```

Comment:
We could also tackle this with the same code:
```
for asset_folder_name in [asset_filename[:30], asset_filename[:250], ""...]:
  ...
```

Reply:
That looks very clean!

Reply:
> Based on a quick google:
> 
> Folder name length limits per OS:
> 
> * Windows 8/10: 255 to 260
> * Mac OS: 1024
> * Linux: 255

One sticking point: That Windows size was for the _total_ length of the **entire** path, all the way from `C:\` to the extension.

That limitation was **finally** lifted in (an _update_ to!) Windows 10, and I believe that change has been automatically pushed to all systems if they've been running Windows Update. But for a long time, Windows path strings were very restricted when you were using legacy APIs (meaning non-UNC, the `//host/share/path/` etc. names.) IIRC that was the reason for the short-ish folder naming. If the user tried to name their project _file_ something over the 260-character limit, the system would stop them, but the worry was that their name might be, say, 258 characters, so adding `_assets` would be the thing that pushed it over.

That being said, if someone is naming their project file something insane and long, and it's accepted as a legal path, these days our extra 7 characters won't often push it over the limit. (Sorry, no... actually I just remembered that it's only _three_ extra characters: the cost of adding `_assets` is offset by the fact that we've dropped the `.osp` extension.)

Except, of course, when `_assets` **is** the straw that breaks 260. There is still a slight chance of causing some headaches for any remaining Win7 users, by allowing significantly longer name lengths. (Which isn't to say that I'm opposed to it, just a factor to consider.)

> But as long as we are truncating the folder name, it seems possible that we can still have a conflict. But perhaps less often.

That's kind of the thing, I mean... if people are going to really push their luck, eventually they're going to break something.

The biggest danger is people creating long-ass project names, then _duplicating_ the files and letting Windows do its `My_long_ass_project_name (1).osp`, `My_long_ass_project_name (Copy).osp`, ... type disambiguation. Because those names only differ at the very far end of the string, any truncation potentially results in a collision.

Reply:
Good point! I forgot the .osp extension. I was going to go with 
`asset_folder_name = asset_filename[:248] + "_assets"` 255 - 7.